implement arrayCompact (#7328)
ClickHouse/ClickHouse: docs/en/query_language/functions/array_functions.md

Synonym for ["arrayReverse"](#array_functions-arrayreverse)

[Original article](https://clickhouse.yandex/docs/en/query_language/functions/array_functions/) <!--hide-->

## arrayCompact(arr) {#array_functions-arraycompact}

> Takes an array, returns an array with elements that are different between two adjacent elements.

Reviewer: ```suggestion
Takes an array, returns an array with consecutive duplicate elements removed.
```

Example:

```sql
SELECT arrayCompact([1, 2, 2, 3, 2, 3, 3])
```

```text

> ┌─arrayDistinct([1, 2, 2, 3, 2, 3, 3])─┐

Reviewer: ```suggestion
┌─arrayCompact([1, 2, 2, 3, 2, 3, 3])─┐
```
